consensus/safety-rules/src/logging.rs

```diff
@@ -1,14 +1,42 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use libra_logger::StructuredLogEntry;
+use crate::Error;
+use consensus_types::common::Round;
+use libra_logger::Schema;
+use libra_types::waypoint::Waypoint;
+use serde::Serialize;
 
-pub fn safety_log(entry: LogEntry, event: LogEvent) -> StructuredLogEntry {
-    StructuredLogEntry::new_named("safety_rules", entry.as_str())
-        .data(LogField::Event.as_str(), event.as_str())
+#[derive(Schema)]
+pub struct SafetyLogSchema {
+    name: LogEntry,
+    event: LogEvent,
+    round: Option<Round>,
```

Comment:
Will `Option` fields not be serialized?  I might have missed that.

Reply:
Yes (I guess more docs are in order). The idea is that, non-optionals will always be logged since they are required, but fields marked as optional will only be logged if it is `Some(T)`